### ISSUE TITLE
MariaDB false positive PRS

### ISSUE THREAD

**Issue by khasanovbi** [sqlfluff/sqlfluff#6265]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Following query
```
INSERT INTO transaction (
    id,
    transaction_status
)
VALUES (
    :id:
    :transaction_status,
)
ON DUPLICATE KEY UPDATE
transaction_status = IF(transaction_status = 'CONFIRMED', transaction_status, VALUES (transaction_status)),
```

return error

### Expected Behaviour

no error

### Observed Behaviour

```
L:  37 | P:  77 |  PRS | Line 37, Position 77: Found unparsable section: ', VALUES (transaction_status)'                                                                                               
WARNING: Parsing errors found and dialect is set to 'mariadb'. Have you configured your dialect correctly?  
```

### How to reproduce

Run sqlfluff for this sql

```
INSERT INTO transaction (
    id,
    transaction_status
)
VALUES (
    :id:
    :transaction_status,
)
ON DUPLICATE KEY UPDATE
transaction_status = IF(transaction_status = 'CONFIRMED', transaction_status, VALUES (transaction_status)),
```

### Dialect

mariadb

### Version

3.1.1

### Configuration

[sqlfluff]
dialect = mariadb
templater = placeholder
output_line_length = 120
warn_unused_ignores = True
sql_file_exts = .sql
max_line_length = 120
exclude_rules = ST07

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:templater:placeholder]
param_style = colon

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @khasanovbi  - as I have mentioned at the previous issue, the above example is not a valid mariadb statement - if you think otherwise, pls point me to the syntax definition which describes this grammar structure / syntax.

**Comment by khasanovbi** [comment]
This query runs on our production on mariadb:11.3.2.
If you don't want to fix bugs, you could just close issue like previous

**Comment by peterbud** [comment]
Minimal repro:

```sql
INSERT INTO table1 VALUES (1,'A') 
  ON DUPLICATE KEY UPDATE
  name=IF(name='A', 'B', VALUES(name));
```

The `VALUES()` function makes it unparsabe. 

See syntax:
https://mariadb.com/docs/server/reference/sql-statements/data-manipulation/inserting-loading-data/insert-on-duplicate-key-update
